src/Microsoft.Health.Fhir.Core/Features/Context/FhirRequestContext.cs

```diff
@@ -55,7 +56,7 @@ public class FhirRequestContext : IFhirRequestContext
 
         public string CorrelationId { get; }
 
-        public Coding RequestType { get; }
+        public ResourceElement RequestType { get; }
```

Comment:
nit: I think we can have our own type instead of using ResourceElement here and just convert it to Coding when needed. I think it was coding because it was easy to use before but now we have to evaluate through FHIR path, it might be easier just to use a ValueTuple or custom struct/class.